shared/domain/src/main/java/com/ivy/domain/usecase/BalanceBuilder.kt

```diff
@@ -0,0 +1,56 @@
+package com.ivy.domain.usecase
+
+import com.ivy.data.model.primitive.AssetCode
+import com.ivy.data.model.primitive.NonZeroDouble
+import com.ivy.data.model.primitive.PositiveDouble
+
+class BalanceBuilder {
+
+    private val balance = mutableMapOf<AssetCode, NonZeroDouble>()
+
+    fun processIncomes(
+        incomes: Map<AssetCode, PositiveDouble>,
+        transferIn: Map<AssetCode, PositiveDouble>,
+    ) {
+        combineAndGetAmount(incomes, transferIn).forEach { (asset, value) ->
+            NonZeroDouble
+                .from((balance[asset]?.value ?: 0.0) + value.value)
+                .onRight { newValue ->
+                    balance[asset] = newValue
+                }
+        }
+    }
+
+    fun processOutcomes(
```

Comment:
Is there a better name than "outcomes"? We should check the banking term for those. Like debits (money out) and credits (money in)

Reply:
I found this link https://www.investopedia.com/terms/w/withdrawal.asp to use for outcomes and "deposit" for incomes in https://www.bankofamerica.com/deposits/manage/glossary/